tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1067,6 +1067,100 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
+			err = utiltc.WaitForTidbClusterCondition(cli, tc.Namespace, tc.Name, time.Minute*10, func(tc *v1alpha1.TidbCluster) (bool, error) {
+				if tc.Status.TiKV.FailoverUID != "" {
+					for _, failureStore := range tc.Status.TiKV.FailureStores {
+						if failureStore.PodName == downPodName {
+							return true, nil
+						}
+					}
+				}
+				return false, nil
+			})
+			framework.ExpectNoError(err, "failed to wait for scale-tikv-1 store to be put into failure stores")
+
+			ginkgo.By("Waiting for scale-tikv-3 pod to be created")
+			newPodName := controller.TiKVMemberName(clusterName) + "-3"
+			err = wait.PollImmediate(time.Second*10, 5*time.Minute, func() (bool, error) {
+				_, err := c.CoreV1().Pods(ns).Get(context.TODO(), newPodName, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				return true, nil
+			})
+			framework.ExpectNoError(err, "failed to wait for scale-tikv-3 pod to be created")
+
+			ginkgo.By("Up scale-tikv-1 store")
+			upPodName := controller.TiKVMemberName(clusterName) + "-1"
+			f.ExecCommandInContainer(upPodName, "tikv", "sh", "-c", "mv /var/lib/tikv/db_down /var/lib/tikv/db")
+
+			ginkgo.By("Waiting for scale-tikv-1 store to be in Up state")
+			err = utiltc.WaitForTidbClusterCondition(cli, tc.Namespace, tc.Name, time.Minute*5, func(tc *v1alpha1.TidbCluster) (bool, error) {
+				for _, store := range tc.Status.TiKV.Stores {
+					if store.PodName == upPodName && store.State == v1alpha1.TiKVStateUp {
+						return true, nil
+					}
+				}
+				return false, nil
+			})
+			framework.ExpectNoError(err, "failed to waiting for scale-tikv-1 store to be in Up state")
+
+			ginkgo.By("Update TiKV failover.recoverByUID")
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Spec.TiKV.RecoverFailover = false
+				tc.Spec.TiKV.Failover = &v1alpha1.Failover{RecoverByUID: tc.Status.TiKV.FailoverUID}
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TiKV failover.recoverByUID")
+
+			ginkgo.By("Waiting for failureStore empty because of recover")
+			err = utiltc.WaitForTidbClusterCondition(cli, tc.Namespace, tc.Name, time.Minute*5, func(tc *v1alpha1.TidbCluster) (bool, error) {
+				if len(tc.Status.TiKV.FailureStores) == 0 && tc.Status.TiKV.FailoverUID == "" {
+					return true, nil
+				}
+				return false, nil
+			})
+			framework.ExpectNoError(err, "failed to wait for failureStore empty because of recover")
+
+			ginkgo.By("Waiting for delete scale-tikv-3 pod because of recover")
+			delPodName := controller.TiKVMemberName(clusterName) + "-3"
+			err = wait.PollImmediate(time.Second*10, 5*time.Minute, func() (bool, error) {
+				_, err := c.CoreV1().Pods(ns).Get(context.TODO(), delPodName, metav1.GetOptions{})
+				if err != nil && apierrors.IsNotFound(err) {
+					return true, nil
+				}
+				return false, nil
+			})
+			framework.ExpectNoError(err, "failed to wait for delete scale-tikv-3 pod because of recover")
```

Comment:
Need to failover again and make sure the new failed Pod cannot be recovered automatically.

Reply:
done